packages/backend/src/core/tx-count/RpcBlockDownloader.ts

```diff
@@ -0,0 +1,67 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+
+import { TxCountRepository } from '../../peripherals/database/TxCountRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { Clock } from '../Clock'
+
+export class RpcBlockDownloader {
+  private taskQueue = new TaskQueue<void>(() => this.update(), this.logger)
```

Comment:
perhpaps this could be renamed to `refreshQueue` or `updateQueue` to make it more obvious what it is